xcell/mappers/mapper_HSC_DR1wl.py

```diff
@@ -0,0 +1,278 @@
+                imsk = 'iflags_pixel_bright'
+                star_mask = np.logical_not(cat[f'{imsk}_object_center'])
+                star_mask *= np.logical_not(cat[f'{imsk}_object_any'])
+                fdfc_mask = cat['wl_fulldepth_fullcolor']
+                shearmask = ishape_flags_mask *\
+                    ishape_sigma_mask *\
+                    ishape_res_mask *\
+                    ishape_shear_mod_mask*star_mask *\
+                    fdfc_mask
+                cat = cat[shearmask]
+
+                # Redshift bin cut
+                zs = cat['pz_best_eab']
+                zbinmask = (zs <= self.z_edges[1]) & (zs > self.z_edges[0])
+                cat = cat[zbinmask]
+
+                # Calibrate shear
+                w = cat[self.w_name]
+                mhat = np.average(cat[f'{isn}_derived_shear_bias_m'],
+                                  weights=w)
+                resp = 1. - np.average(cat[f'{isn}_derived_rms_e'] ** 2,
+                                       weights=w)
+                e1 = (cat[f'{isn}_e1']/(2.*resp) -
+                      cat[f'{isn}_derived_shear_bias_c1']) / (1 + mhat)
+                e2 = (cat[f'{isn}_e2']/(2.*resp) -
+                      cat[f'{isn}_derived_shear_bias_c2']) / (1 + mhat)
+                cat['e1'] = e1
+                cat['e2'] = e2
+
+                # Remove unnecessary columns
+                cat.keep_columns(['ra', 'dec', 'e1', 'e2', self.w_name])
+                cats.append(cat)
+            self.cat = vstack(cats)
+            if fname_lite is not None:
+                self.cat.write(fname_lite)
+        return self.cat
+
+    def _clean_raw_catalog(self, fnames):
+        cats = []
+        for fname in fnames:
+            if not os.path.isfile(fname):
+                raise ValueError(f"File {fname} not found")
+            c = Table.read(fname)
+            sel = np.ones(len(c), dtype=bool)
+            isnull_names = []
+            for key in c.keys():
+                if key.__contains__('isnull'):
+                    if not key.startswith('ishape'):
+                        sel[c[key]] = 0
+                    isnull_names.append(key)
+                else:
+                    # Keep photo-zs and shapes even if they're NaNs
+                    if ((not key.startswith("pz_")) and
+                            (not key.startswith('ishape'))):
+                        sel[np.isnan(c[key])] = 0
+            c.remove_columns(isnull_names)
+            c.remove_rows(~sel)
+
+            # Collect sample cuts
+            sel_area = c['wl_fulldepth_fullcolor']
+            sel_clean = sel_area & c['clean_photometry']
+            sel_maglim = np.ones(len(c), dtype=bool)
+            sel_maglim[c['icmodel_mag'] -
+                       c['a_i'] > self.icut] = 0
+            # Blending
+            sel_blended = np.ones(len(c), dtype=bool)
+            # abs_flux<10^-0.375
+            sel_blended[c['iblendedness_abs_flux'] >= 0.42169650342] = 0
```

Comment:
It'd be great to have a comment here with the reference to where these cuts come from. I couldn't find the numbers in the HSC paper in a quick search.

Reply:
Added

Reply:
Thanks!